### ISSUE TITLE
Odd error message when approving comment in panel

### ISSUE THREAD

**Issue by sebastiangreger** [sebastiangreger/kirby3-commentions#57]
I'm not quite sure where to look - I am suddenly getting this error when trying to approve/unapprove comments in the panel:

![Screenshot from 2020-05-14 23-49-01](https://user-images.githubusercontent.com/6355217/81989586-cde91f00-963d-11ea-991c-922ef7c01446.png)

I ran the parcel build process again, so the JS file should be well formed. Any idea is this something with the code or a problem on my install, @fabianmichael?

**Comment by fabianmichael** [comment]
Seems to be linked to Babel, the JavaScript transpiler that Parcel uses, maybe it picked up a babel config file from a parent directory?

I would try to delete node_modules and re-install first, if that fails:

https://stackoverflow.com/questions/33527653/babel-6-regeneratorruntime-is-not-defined

**Comment by sebastiangreger** [comment]
Tried various re-installs etc. but could not pinpoint any local environment issues.

Your hint about Babel led me to this for Babel 7: https://stackoverflow.com/a/54490329 ...by adding those two `import` statements (https://github.com/sebastiangreger/kirby3-commentions/commit/4571a14276a9bb709d3f8c3bce8ae17294c99e6e) the error is gone. So it appears this is related to `async` (`updateCommention` and `deleteCommention` are the two only functions in the compiled script that reference `regeneratorRuntime`) and lack of support without importing these extra dependencies?

So problem solved for now :balloon:

Only downside: the compiled index.js is now a whopping 116kB compared to 11kB earlier :scream:

**Comment by sebastiangreger** [comment]
Update, as I am still wondering about this:

In another project, where I [reuse large parts of `Panel.vue`](https://github.com/sebastiangreger/kirby3-sendmentions/blob/1b7edbfbaa28c137dcd7f7efae55001bbe202c13/src/components/Section.vue#L93) (including the `async` API call that appeared to be problematic in Commentions), the code compiles and runs just fine without adding those additional imports. So maybe something about local setup after all...

**Comment by fabianmichael** [comment]
@sebastiangreger I got the same error today, but after re-building it was gone. Strange …

But on the other hand, if using `async` it the root of the issue, it should not be to hard to rewrite the sections without using `aysnc`/`await`. Should we try to do that?

**Comment by sebastiangreger** [comment]
This is an absolute mystery to me. What happens if you take out [the two `import` statements from `src/index.js`](https://github.com/sebastiangreger/kirby3-commentions/blob/329841c0ccb76b99a479e85b2a9ccaf4a4284103/src/index.js#L3) and re-build? Do you get the error then?

I'm just wondering was it just gone because I had added those two statements to the repo earlier? Or does this really just occur when building on my machine...

What really boggles my mind is the fact that I use the verbatim same `async/await` code in the other plugin and there I do not get the error. So either something is severely wrong with my dev setup for the commention plugin specifically (I'm running out of ideas what it could be), or there is something in our `index.js` that causes this problem?!

As said: the inclusion of the two import statements "solved" it for now, but the size of that complied JS file is rather, ahem, excessive :see_no_evil:

**Comment by sebastiangreger** [comment]
In other news: I seem to develop a tendency to click the wrong "Comment" button :rofl:

**Comment by fabianmichael** [comment]
Mhh … that’s strange. Tried again without the two imports and it still worked fine. My setup:

- Parcel 1.12.4
- Node 12.16.2

The only thing that comes into mind, is like I said earlier, that this could be related to some config file in a parent directory. Could be a browserlist-config, that includes legacy browsers as compile-target or similar. What happens, when you clone a copy of the repo into a different folder, that’s not part of any real project and try to build from there?

**Comment by sebastiangreger** [comment]
This is **very** odd. My setup features the same Parcel/Node versions, but even when cloning the repo into my home folder directly, I still get the exact same error after building it. Also, any parent folders above my regular plugin folder are completely void of anything Node-related, so I can't quite see where this would come from (also tried deleting/reinstalling various Node packages, caches etc. - anything I could think of).

Trying to troubleshoot this already consumed waaaay too much of my time and I don't have any clue where it may come from, so have to put this to rest :yawning_face: ...after a bit of experimentation, I eventually removed the `core-js` dependency from `src/index.js`, only leaving the `import "regenerator-runtime/runtime";` line out of the two additions I made earlier - this now only adds 7kB (instead of 110kB) extra to the final package, and I have lost any motivation to further research this...

The error is gone - problem solved! :hear_no_evil: :see_no_evil:
